src/input.rs

```diff
@@ -1354,6 +1362,94 @@ impl State {
             },
         );
     }
+
+    /// Computes the cursor position for the touch event.
+    ///
+    /// This function handles the touch output mapping, as well as coordinate transform
+    fn compute_touch_location<I: InputBackend, E: AbsolutePositionEvent<I>>(
+        &self,
+        evt: &E,
+    ) -> Option<Point<f64, Logical>> {
+        let output = self.niri.output_for_touch()?;
+        let output_geo = self.niri.global_space.output_geometry(output).unwrap();
+        let transform = output.current_transform();
+        let size = transform.invert().transform_size(output_geo.size);
+        Some(
+            transform.transform_point_in(evt.position_transformed(size), &size.to_f64())
+                + output_geo.loc.to_f64(),
+        )
+    }
+
+    fn on_touch_down<I: InputBackend>(&mut self, evt: I::TouchDownEvent) {
+        let Some(handle) = self.niri.seat.get_touch() else {
+            return;
+        };
+        let Some(touch_location) = self.compute_touch_location(&evt) else {
+            return;
+        };
+        let serial = SERIAL_COUNTER.next_serial();
+        let under = self
+            .niri
+            .surface_under_and_global_space(touch_location)
+            .map(|under| under.surface);
```

Comment:
We probably want to activate window / output under touch here, just like we do for mouse and tablet clicks?

Reply:
I initially intended on adding that through bindings on global gestures. But I guess for now we can just active the window similar to pointer button. One important difference is that I added a check if there is a touch grab and in this case not activate a window. I added that to prevent stealing the window on multiple touches.